### PR TITLE
refacto of the attachement.csv import => no more crash or latencies

### LINKED ISSUE #332: Data model optimization for filtering
**Ticket to be separate**

The recent tests showed that injecting large (30 000 rows x 6 columns) csv data into the filters are causing application unstable. This is because the filters are in memory. This has to be filtered in indexdb.

To be decided technically: 
- document existing datamodel (Martin)
- project a new datamodel that fits the needs (?)
- design the new strategy 
- implement

There is scripts that preprocess datas, they have to be included if public else saved in a private repository

### PR DESCRIPTION
**Pull request by PascalSeniziRTE**
### CSV import pipeline — performance refactor & generalization  [#332](https://github.com/phlowers/phlowers-stellar-app/issues/332)

This PR fixes a critical performance issue on `attachments.csv` import and then generalizes the new streaming pipeline to all six catalog CSVs.

---

### 1. `attachments.csv` — eliminate main-thread crash & UI latency (commit `6ce02c3`)

**Problem.** Importing the production `attachments.csv` (~25 MB / ~230 000 rows) loaded the entire file into memory via `Papa.parse({ complete })`, mapped every row to a flat IndexedDB record, then called a single `bulkAdd`. This caused:
- Main-thread memory spikes of ~280 MB.
- UI freezes of several seconds.
- Occasional out-of-memory crashes on lower-end machines.

**Fix.**
- New grouped schema `catSupportAttachments` (Dexie **V6**): one row per `support_name` instead of one row per attachment set, drastically reducing the B-tree size (~230 k flat rows → ~500 grouped rows).
- New dedicated Web Worker (`attachment-import.worker.ts`) that streams the CSV through PapaParse chunk-by-chunk and merges groups into IndexedDB incrementally via `bulkGet` + `bulkPut`.
- Backpressure on each chunk (`parser.pause()` / `parser.resume()`) keeps memory flat.
- `AttachmentService` refactored to expose the grouped API; legacy flat methods documented in deadcode.md.

**Result.** Main-thread memory stays under ~10 MB during import, no more freezes or crashes.

---

### 2. Generalize the streaming pipeline to all 6 catalog CSVs (commit `302e675`)

After the attachments fix proved the approach, the same architecture is now mutualized for every catalog import (`attachments`, `cables`, `chains`, `lines`, `maintenance`, `obstacles`).

**New shared module:** csv-import
- **csv-import.engine.ts** — pure, framework-agnostic engine that streams a CSV, drives a config, and emits `progress` / `done` / `error` messages. Fully unit-tested with injected dependencies (Papa, table resolver, clock).
- **csv-import.worker.ts + `internal/run-worker-import.ts`** — generic Web Worker entry point; the runtime logic lives in a separate module so unit tests can import it under jsdom without registering a top-level `message` listener.
- **csv-import.client.ts** — Angular service (`CsvImportClientService`) that spawns the worker, forwards progress, and terminates on done/error.
- **`configs/`** — one declarative `CsvImportConfig` per catalog (row mapping, delimiter, optional `finalize` hook for dedup-then-write patterns like `lines.csv`). Registered in `CSV_IMPORT_REGISTRY`.
- **csv-import.worker-db.ts** — opens the same Dexie database from the worker; reuses the new shared **`applyStellarDbVersions(db)`** helper so the worker and the main thread can never drift apart.

**Refactored services** (`AttachmentService`, `CablesService`, `ChainsService`, `LinesService`, `MaintenanceService`, `ObstaclesService`):
- Public API unchanged — `importFromFile()` now delegates to `CsvImportClientService.importCsv(<key>)`.
- The legacy `attachment-import.worker.*` files (worker, interfaces, spec) are removed and logged as entry #30 in deadcode.md.

**Documentation:**
- New README.md in the `csv-import/` folder with the architecture diagram and a 3-step "how to add a new CSV" checklist.
- TSDoc inline `{@link X}` tags converted to `` `X` `` across the codebase to fix Sphinx/sphinx-js documentation build (`Not implemented` runtime error on inline tags).

---

### Tests

- **+91 new unit tests** (Vitest only, no Jest): engine (18), client (6), worker (3), per-config specs (40), cell helpers (20), registry (4).
- **6 anonymized CSV fixtures** added under csv (with README.md) covering all edge cases: empty keys, French decimals, duplicates, semicolon delimiter, fallback IDs.
- All existing service specs rewritten compactly (mocking `CsvImportClientService` instead of the old worker), preserving full read-side coverage.
- **Full test suite: 136 files / 3119 tests passing, 0 lint errors.**

---

### Migration notes

- Dexie V6 is additive: `catSupportAttachments` is created and `catAttachments` is dropped automatically on first launch — no manual migration required.
- All public service APIs are byte-for-byte identical — no caller changes needed.

### CODE REVIEW

File: src/app/shared/catalog/csv-import/configs/cables.config.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    is_polynomial: item.is_polynomial === 'true' || item.is_polynomial === 'True',
+    diameter_heart: convertStringToNumber(item.diameter_heart),
+    section_conductor: convertStringToNumber(item.section_conductor),
+    section_heart: convertStringToNumber(item.section_heart),
+    solar_absorption: convertStringToNumber(item.solar_absorption),
+    emissivity: convertStringToNumber(item.emissivity),
+    electric_resistance_20: convertStringToNumber(item.electric_resistance_20),
+    linear_resistance_temperature_coef: convertStringToNumber(item.linear_resistance_temperature_coef),
+    radial_thermal_conductivity: convertStringToNumber(item.radial_thermal_conductivity),
+    has_magnetic_heart: item.has_magnetic_heart === 'true',
```
Comment (pikr-fra-RTE) [inline src/app/shared/catalog/csv-import/configs/cables.config.ts]: There's no "True" trap for this one in CSV files?
Bonus point: I see a toOptionalBoolean method, API or function just below. Is there a toBoolean() that could take care of this and "is_polynomial" l.37?

File: src/app/shared/catalog/csv-import/configs/lines.config.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    uuid: uuidv4(),
+    link_idr: item.link_idr || '',
+    link_adr: item.link_adr || '',
+    lit_idr: item.lit_idr || '',
+    lit_adr: item.lit_adr || '',
+    branch_id: item.branch_id || '',
+    branch_idr: item.branch_idr || '',
+    branch_adr: item.branch_adr || '',
+    voltage_idr: hasNoVoltage ? $localize`NO VOLTAGE` : item.voltage_idr,
+    voltage_adr: hasNoVoltage ? 'NO_VOLTAGE' : item.voltage_adr
```
Comment (pikr-fra-RTE) [inline src/app/shared/catalog/csv-import/configs/lines.config.ts:28]: Inconsistency between IDR and ADR.
IDR seems to be a displayed string (hence the localize) while ADR looks like a technical name.

Is this normal behavior? Does ADR would require to be translated too?

**Review by pikr-fra-RTE — CHANGES_REQUESTED** [review]

File: src/app/shared/catalog/csv-import/configs/lines.config.ts (hunk 2 above)
Comment (PascalSeniziRTE) [reply src/app/shared/catalog/csv-import/configs/lines.config.ts:28]: Normal ? For sure ! 😎

**Review by pikr-fra-RTE — APPROVED** [review]
